Indicators/RelativeDailyVolume.cs

```diff
@@ -0,0 +1,141 @@
+        /// </summary>
+        /// <param name="period">The period over which to perform to computation</param>
+        public RelativeDailyVolume(int period = 14)
+            : this($"RDV({period})", period)
+        {
+        }
+        /// <summary>
+        /// Creates a new RelativeVolume indicator with the specified period
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period of this indicator</param>
+        public RelativeDailyVolume(string name, int period)
+            : base(name)
+        {
+            _previousDay = -1;
+            WarmUpPeriod = period;
+            _rollingTime = new RollingWindow<DateTime>(period * 2);
+            _rollingData = new RollingWindow<decimal>(period * 2);
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// </summary>
+        /// <param name="input">The input value to this indicator on this time step</param>
+        /// <returns>A a value for this indicator</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            if (input.Time.Day != _previousDay)
+            {
+                _currentVolume.Clear();
+                _previousDay = input.Time.Day;
+            }
+
+            _currentVolume.Add(input.Volume);
+            _historicalVolumes.Add(input.Time, input.Volume);
+            _rollingTime.Add(input.Time);
+            _rollingData.Add(input.Volume);
+
+            if (!IsReady) { return 0; }
+
+            var todaysTotal = _currentVolume.Sum();
+            var timeList = _rollingTime.ToList();
+            var dataList = _rollingData.ToList();
+            int listCount = timeList.Count;
+            var currentDay = -1;
+            int listDifference = (listCount - WarmUpPeriod) - 1;
+            decimal cummulateValue = 0;
+            var relativeValues = new List<decimal>();
+
+            if (listDifference != -1) { timeList.Skip(listDifference); }
+            foreach (var pair in timeList)
+            {
+                if (currentDay != pair.Day)
+                {
+                    if (cummulateValue != 0) { relativeValues.Add(cummulateValue); }
+                    cummulateValue = 0;
+                }
+
+                var dayDelta = input.Time - pair;
+                if ((dayDelta.Days >= 1) && (pair.Hour <= input.Time.Hour) && (pair.Minute <= input.Time.Minute) && (input.Time.Second <= pair.Second))
+                {
+                    cummulateValue += dataList[timeList.IndexOf(pair)];
+                }
+                currentDay = pair.Day;
+            }
+
+            if (cummulateValue != 0) { relativeValues.Add(cummulateValue); }
+
+            if (relativeValues.Count == 0)
+            {
+                throw new ArgumentException("Need atleast one day of data", nameof(WarmUpPeriod));
```

Comment:
Just adding it to the basic template algorithm throws this exception (`atleast` typo), don't think this is correct behavior.

Seems what would work here is holding a SMA instance per time of a day? Each new data point we get the SMA to use and once we get it's value we can update it with the current cumulative volume?

Reply:
It would be difficult to use an SMA as the period for the SMA would be unknown. Instead, I moved the ArgumentException to where the indicator is initialized. Please reference the method: CheckLength.

Reply:
> It would be difficult to use an SMA as the period for the SMA would be unknown.

Isn't the period of the RDV the amount of days, I'd expect it to be? If not, what is the period representing?